### PR TITLE
Update README: Include deploy and remove in list of commands, readability and consistency updates

### PR DESCRIPTION
**Pull request by fordN**
Updates to README:
  - Include `deploy` and remove in list of commands. 
  - Change step 8 to `yarn build`, step 9 already describes `deploy.`
  - Update `How It Works` bullet formatting to be cleaner and more readable.
  - Mention deployment to node in `How It Works.`
  - Use <> to mark arguments in sample commands that should be replaced by user values.
  - Recommend using `yarn` for generation and building rather than `npm`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,27 +5,31 @@
 
 The Graph command line interface.
 
-As of today, the command line interface consists of two commands:
+As of today, the command line interface consists of four commands:
 
 - `graph codegen` — generates TypeScript code for smart contract ABIs used in subgraphs.
 - `graph build` — compiles subgraphs to WebAssembly and deploys them to IPFS.
+- `graph deploy` — deploys subgraphs to running [Graph Nodes](https://github.com/graphprotocol/graph-node).
```
[the remark this answers is not in the document]
Comment (nenadjaja) [reply README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -5,25 +5,26 @@
...
 ## How It Works
 
-`graph` takes a subgraph manifest (defaults to `subgraph.yaml`) with
+`graph` takes a subgraph manifest (defaults to `subgraph.yaml`) with references to
 
-- references to a GraphQL schema,
-- smart contract ABIs, and
-- mappings written in TypeScript/AssemblyScript,
+- a GraphQL schema,
+- smart contract ABIs
```
Comment (Jannis) [inline README.md]: Inconsistent commas here. Use a comma at the end of each bullet point and a full stop at the end of the last one.

File: README.md [hunk 3]
```diff
@@ -98,9 +99,17 @@ For clarity, an example of the setup below can be found in the [ENS subgraph rep
        --verbosity debug \
        --node http://127.0.0.1:8020/ \
        --ipfs /ip4/127.0.0.1/tcp/5001 \
-       --subgraph-name your-subgraph
+       --subgraph-name <SUBGRAPH_NAME>
     ```
 
+If you need to remove a subgraph from the [Graph Node](https://github.com/graphprotocol/graph-node) use:
```
Comment (Jannis) [inline README.md]: Simplify to:
```
To remove a subgraph from the [Graph Node](...), use:
```

Comment (Jannis) [reply README.md]: (Note the comma as well)

File: README.md [hunk 4]
```diff
@@ -98,9 +99,17 @@ For clarity, an example of the setup below can be found in the [ENS subgraph rep
...
+```sh
+graph \
+  remove \
+  --api-key <KEY> \
```
Comment (Jannis) [inline README.md]: This is optional, so let's not mention it.

File: README.md (hunk 2 above)
Comment (Jannis) [reply README.md]: Damn, you're right. Yes, personally I'd start a new sentence after the last bullet point, but can't claim I'm consistent with that.

Comment (Jannis) [reply README.md]: I sometimes even use Oxford comma between the last two bullet points, like so:
```
- foo,
- bar, and
- baz.
```
I'm weird.

**Comment by Jannis** [comment]
@fordN Looks great now! All it needs is a rebase.

**Pull request merged**